packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,376 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { ContainerOverride } from '..';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Basic properties for ECS Tasks
+ */
+export interface EcsRunTaskProps extends sfn.TaskStateBaseProps {
+  /**
+   * The ECS cluster to run the task on
+   */
+  readonly cluster: ecs.ICluster;
+
+  /**
+   * [disable-awslint:ref-via-interface]
+   * Task Definition used for running tasks in the service.
+   *
+   * Note: this must be TaskDefinition, and not ITaskDefinition,
+   * as it requires properties that are not known for imported task definitions
+   */
+  readonly taskDefinition: ecs.TaskDefinition;
+
+  /**
+   * Container setting overrides
+   *
+   * Key is the name of the container to override, value is the
+   * values you want to override.
+   *
+   * @default - No overrides
+   */
+  readonly containerOverrides?: ContainerOverride[];
+
+  /**
+   * In what subnets to place the task's ENIs
+   *
+   * @default - Private subnet if assignPublicIp, public subnets otherwise
+   */
+  readonly subnets?: ec2.SubnetSelection;
+
+  /**
+   * Existing security group to use for the tasks
+   *
+   * @default - A new security group is created
+   */
+  readonly securityGroup?: ec2.ISecurityGroup;
+
+  /**
+   * Assign public IP addresses to each task
+   *
+   * @default false
+   */
+  readonly assignPublicIp?: boolean;
+
+  /**
+   * An Amazon ECS launch type determines the type of infrastructure on which your
+   * tasks and services are hosted.
+   *
+   * @see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/launch_types.html
+   */
+  readonly launchTarget: IEcsLaunchTarget;
+}
+
+/**
+ * An Amazon ECS launch type determines the type of infrastructure on which your tasks and services are hosted.
+ * @see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/launch_types.html
+ */
+export interface IEcsLaunchTarget {
+  /**
+   * called when the ECS launch target is configured on RunTask
+   */
+  bind(_task: EcsRunTask): EcsLaunchTargetConfig;
+}
+
+/**
+ * Configuration options for the ECS launch type
+ */
+export interface EcsLaunchTargetConfig {
+  /**
+   * Additional parameters to pass to the base task
+   *
+   * @default - No additional parameters passed
+   */
+  readonly parameters?: { [key: string]: any };
+}
+
+/**
+ * Properties to define an ECS service
+ */
+export interface EcsFargateLaunchTargetOptions {
+  /**
+   * Fargate platform version to run this service on
+   *
+   * Unless you have specific compatibility requirements, you don't need to
+   * specify this.
+   *
+   * @default FargatePlatformVersion.LATEST
```

Comment:
I think you have a reasonable point here still. It is done in `ecs-patterns` but patterns can be opinionated.

another thing to [consider] (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/platform_versions.html#platform-version-considerations)

`LATEST` currently maps to `1.3.0`
`1.4.0` has been out since May 28, 2020 but is not `LATEST` yet

what would you recommend here - default of `1.4.0` or `LATEST` ? copy @MrArnoldPalmer

Reply:
This is a good question, I’d like to ask @SoManyHs about this. Should CDK generally be using ‘latest’ for platform version?

Reply:
If they **guarantee** backwards compat, this will be ok. 

In ECS, a production service marked LATEST is likely to be configured once and never deleted, so it's probably pinned to the 'LATEST' at creation time and stays the same. The docs seem to indicate that.

However, with StepFunctions, I suspect that the LATEST will be re-calculated every time an ECS task is executed in any state machine. 

@shivlaks - you could unblock yourself here by making this a mandatory property and changing it to optional when we have better answers. (changing a property to optional is backwards compat, but changing its default is not)

Reply:
that's a great suggestion @nija-at 
I went ahead and made it required. that way we're not boxed into a corner and we can offer a sensible default (which may still turn out to be `LATEST`) without breaking users.